lightning/src/routing/router.rs

```diff
@@ -192,10 +195,14 @@ pub struct Payee {
 
 	/// Expiration of a payment to the payee, in seconds relative to the UNIX epoch.
 	pub expiry_time: Option<u64>,
+
+	/// The maximum total CLTV delta we accept for the route.
+	pub max_total_cltv_expiry_delta: Option<u32>,
 }
 
 impl_writeable_tlv_based!(Payee, {
 	(0, pubkey, required),
+	(1, max_total_cltv_expiry_delta, option),
```

Comment:
Hum, I have no strong opinion on this, but making it non-optional would  probably require that `max_total_cltv_expiry_delta` is explicitly given with each call of `from_node_id(..)` and `for_keysend(..)`, right? 
While implementing the changes I found that making it optional and using the default value when no other value is set would be way cleaner, especially it would not require to know about, import,  and use `DEFAULT_MAX_TOTAL_CLTV_EXPIRY_DELTA` all over the place.

Reply:
> Hum, I have no strong opinion on this, but making it non-optional would probably require that max_total_cltv_expiry_delta is explicitly given with each call of from_node_id(..) and for_keysend(..), right?

I'd think we'd just set it to the default value - that is, after all, the implicit behavior in the router - it uses the default value if none is set.

Reply:
Before I'd scrolled to that point in the code, when I saw the value was an `Option`, my assumption was that, if unset, no limit would be enforced, which feels intuitive for a `maximum_*: Option<..>` field. Given its still enforced (which feels like the correct behavior), it seems strange the field is an option.

Reply:
Ah, I see. Done in https://github.com/lightningdevkit/rust-lightning/pull/1234/commits/689518520e9daadb7fddd2b836bc19815ef24921.